multimodal/src/autogluon/multimodal/optimization/losses.py

```diff
@@ -272,3 +272,71 @@ def forward(self, features_a, features_b, logit_scale, rank=0, world_size=1):
 
         total_loss = (F.cross_entropy(logits_per_a, labels) + F.cross_entropy(logits_per_b, labels)) / 2
         return total_loss
+
+
+class FocalLoss(nn.Module):
```

Comment:
Can we use the torchvision's implementation?https://pytorch.org/vision/main/generated/torchvision.ops.sigmoid_focal_loss.html

Reply:
We have torchvision installed.

Reply:
oh that would be much nicer

Reply:
is the torchvision implementation for binary classification? I saw in their source code
```
p = torch.sigmoid(inputs)
    ce_loss = F.binary_cross_entropy_with_logits(inputs, targets, reduction="none")
    p_t = p * targets + (1 - p) * (1 - targets)
    loss = ce_loss * ((1 - p_t) ** gamma)

    if alpha >= 0:
        alpha_t = alpha * targets + (1 - alpha) * (1 - targets)
        loss = alpha_t * loss
```

Reply:
I find this post had a discussion on the binary/multiclass cases. 
https://github.com/pytorch/vision/issues/3250

Reply:
oh okay. I feel like based on the discussion we'll still need to modify the code manually?

Reply:
Yes, we can keep this implementation here.